crypto/keys/ecdsa/ecdsa_privkey.go

```diff
@@ -0,0 +1,129 @@
+package ecdsa
+
+import (
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/sha256"
+	"fmt"
+	"math/big"
+
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	gogotypes "github.com/gogo/protobuf/types"
+)
+
+// "github.com/cosmos/cosmos-sdk/codec"
+// "github.com/cosmos/cosmos-sdk/types/errors"
+
+type ecdsaSK struct {
+	ecdsa.PrivateKey
+}
+
+// GenSecp256r1 generates a new secp256r1 private key. It uses OS randomness.
+func GenSecp256r1() (cryptotypes.PrivKey, error) {
+	key, err := ecdsa.GenerateKey(secp256r1, rand.Reader)
+	return &ecdsaSK{*key}, err
+}
+
+// PubKey implements SDK PrivKey interface
```

Comment:
What SDK? Cosmos SDK? Android SDK? Please use full, correct names for referencing functions, types, etc.

Reply:
And please, terminate your sentences with full stop.

Reply:
It seams that there is no single style for terminating comments - in most of the places we don't put `.`. But I will add it.